librz/core/ctypes.c

```diff
@@ -1026,22 +1026,78 @@ RZ_IPI bool rz_types_open_file(RzCore *core, const char *path) {
 	return true;
 }
 
-RZ_IPI void rz_types_open_editor(RzCore *core, const char *typename) {
+RZ_IPI bool rz_types_open_editor_all(RzCore *core) {
 	RzTypeDB *typedb = core->analysis->typedb;
-	char *str = rz_core_cmd_strf(core, "tc %s", typename ? typename : "");
-	char *tmp = rz_core_editor(core, "*.h", str);
-	if (tmp) {
-		char *error_msg = NULL;
-		int result = rz_type_parse_string_stateless(typedb->parser, tmp, &error_msg);
-		if (result) {
-			// TODO: remove previous types and save new edited types
-			//rz_type_db_purge(typedb);
-		}
-		if (error_msg) {
-			eprintf("%s\n", error_msg);
-			free(error_msg);
+	RzStrBuf *contents = rz_strbuf_new("");
+	if (!contents) {
+		return false;
+	}
+	// Get the all types
+	RzList *types = rz_type_db_get_base_types(typedb);
```

Comment:
Does it make sense to list the atomic types like char/int/etc?

Reply:
Do we need typedefs? Or only enums, structures and unions?

Reply:
I think also typedefs make sense